test/components/collision_detection_test.dart

```diff
@@ -0,0 +1,550 @@
+import 'package:flame/geometry.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/geometry.dart' as geometry;
+import 'package:flame/src/geometry/circle.dart';
+import 'package:flame/src/geometry/line_segment.dart';
+import 'package:flame/src/geometry/line.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('LineSegment.isPointOnSegment tests', () {
+    test('Can catch simple point', () {
+      final segment = LineSegment(
+        Vector2.all(0),
+        Vector2.all(1),
+      );
+      final point = Vector2.all(0.5);
+      assert(segment.containsPoint(point), "Point should be on segment");
```

Comment:
1. don't we use `expect` on tests? assert just throws an exception right?
2. single quotes
(this applies to all tests)

Reply:
Aaah yes, I have to change to expect on all tests, don't remember why I made them asserts.

Reply:
Fixed